Lib/test/test_array.py

```diff
@@ -1205,6 +1216,15 @@ def test_issue17223(self):
         self.assertRaises(ValueError, a.tounicode)
         self.assertRaises(ValueError, str, a)
 
+    def test_typecode_u_deprecation(self):
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always", DeprecationWarning)
+            array.array("u")
+        self.assertGreaterEqual(len(w), 1)
+        for warning in w:
+            self.assertIs(warning.category, DeprecationWarning)
```

Comment:
Can't you use [`assertWarns`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns) here?

Reply:
Ah yes, much better!